runtime/src/accounts.rs

```diff
@@ -264,6 +266,15 @@ impl Accounts {
             } else {
                 LoadZeroLamports::SomeWithZeroLamportAccount
             };
+        let mut accumulated_accounts_data_size = 0u64;
```

Comment:
Note that we do not count the data size of accounts that are "loader keys" (line 298) because they are loaded later in `load_executable_accounts`. You'll need to make sure to track data size in `load_executable_accounts` as well. However, it's tricky because non "loader keys" could be double counted. A "loader key" is defined as a key which is invoked but not included passed as an account input to any instructions. So it's possible that a program account loaded in `load_executable_accounts` was also loaded in `load_transaction` because it was considered a "non-loader key" and then we would double count the data size of the program and its programdata account.

Reply:
Again, thanks for pointing it out! Will follow it through.

Reply:
commit de60af5da7a054e7199fc91dcdbaa5297106a3d1 added logic to avoid double counting non-loader if reloaded in load_executable_account().